src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -912,17 +914,17 @@ def perturb_individual_age(self, pop: pd.DataFrame) -> pd.Series:
 
             # Calculate shifted ages and see if any are negative and need to be resampled.
             shifted_ages = pop.loc[to_shift, "age"] + age_shift
-            negative_ages_mask = shifted_ages < 0
-            pop.loc[~negative_ages_mask & to_shift, "age"] = shifted_ages.loc[
-                ~negative_ages_mask & to_shift
-            ]
-            to_shift = negative_ages_mask
+            positive_ages_idx = shifted_ages.loc[shifted_ages >= 0].index
+            pop.loc[positive_ages_idx, "age"] = shifted_ages.loc[positive_ages_idx]
+            to_shift = shifted_ages.loc[
```

Comment:
I think this can just be `to_shift = shifted_ages.index.difference(positive_ages_idx)`